src/Nethermind/Nethermind.Blockchain/TransactionPools/Filters/AcceptWhenTransactionFilter.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Dirichlet.Numerics;
+
+namespace Nethermind.Blockchain.TransactionPools.Filters
+{
+    public class AcceptWhenTransactionFilter : ITransactionFilter
+    {
+        private readonly Builder.Filter _addFilter;
+        private readonly Builder.Filter _deleteFilter;
+
+        private AcceptWhenTransactionFilter(Builder.Filter addFilter, Builder.Filter deleteFilter)
+        {
+            _addFilter = addFilter;
+            _deleteFilter = deleteFilter;
+        }
+
+        public bool CanAdd(Transaction transaction) => Valid(transaction, _addFilter);
+        public bool CanDelete(Transaction transaction) => Valid(transaction, _deleteFilter);
+
+        private static bool Valid(Transaction transaction, Builder.Filter filter)
```

Comment:
why is it accepting Builder.Filter? Builder seems to be a test namespace

Reply:
It's a nested class within Filter itself, however, the name might be confusing.